internal/service/settler/settler.go

```diff
@@ -127,32 +140,28 @@ func (s *Server) constructSettlementData(ctx context.Context, epoch uint64, curs
 	}, nil
 }
 
-// calculateOperationRewards calculates the Operation Rewards for all Nodes
-// For Alpha, there is no actual calculation logic, the rewards are set to 0
-// TODO: Implement the actual calculation logic
-func calculateOperationRewards(nodes []common.Address) []*big.Int {
-	slice := make([]*big.Int, len(nodes))
-
-	// For Alpha, set the rewards to 0
-	for i := range slice {
-		slice[i] = big.NewInt(0)
+// updateNodeStakingData retrieves node information from a staking contract
+// and updates the staking pool tokens for each node.
+func (s *Server) updateNodeStakingData(nodeAddresses []common.Address, nodes []*schema.Node) error {
+	nodeInfo, err := s.stakingContract.GetNodes(&bind.CallOpts{}, nodeAddresses)
+	if err != nil {
+		return fmt.Errorf("get nodes from chain: %w", err)
 	}
 
-	return slice
-}
-
-// prepareRequestCounts prepares the Request Counts for all Nodes
-// For Alpha, there is no actual calculation logic, the counts are set to 0
-// TODO: Implement the actual logic to retrieve the counts from the database
-func prepareRequestCounts(nodes []common.Address) []*big.Int {
-	slice := make([]*big.Int, len(nodes))
+	nodeInfoMap := lo.SliceToMap(nodeInfo, func(node l2.DataTypesNode) (common.Address, l2.DataTypesNode) {
+		return node.Account, node
+	})
 
-	// For Alpha, set the counts to 0
-	for i := range slice {
-		slice[i] = big.NewInt(0)
+	for _, node := range nodes {
+		if nodeInfo, ok := nodeInfoMap[node.Address]; ok {
+			stakePoolTokens := nodeInfo.StakingPoolTokens
+			tokenDecimals := big.NewInt(1e18)
+			stakePoolTokens.Div(stakePoolTokens, tokenDecimals)
```

Comment:
It's better to keep 18 decimals here I think, because it is handled this way elsewhere as well